PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/ui/SparkSubmissionJobUploadStorageCtrl.kt

```diff
@@ -126,27 +132,46 @@ abstract class SparkSubmissionJobUploadStorageCtrl(val view: SparkSubmissionJobU
     }
 
     fun selectCluster(clusterName: String) {
-        view.storageCheckSubject.onNext("Selected cluster $clusterName")
+        view.storageCheckSubject.onNext("$selectClusterMsg $clusterName")
```

Comment:
If the `storageCheckSubject` event is used, it might be better to change it to typed messages, rather than `String`

Reply:
```kt
    open class StorageCheckEvent(val message: String)
    class StorageCheckClusterSelectedEvent(val clusterName: String) : StorageCheckEvent("Selected cluster $clusterName")
    class StorageCheckSignInOutEvent(): StorageCheckEvent("After user clicked.....")

    val storageCheckSubject: PublishSubject<StorageCheckEvent> = PublishSubject.create()
```